src/GraphQL.Client/Websocket/GraphQLTransportWSProtocolHandler.cs

```diff
@@ -242,7 +240,7 @@ public IObservable<GraphQLResponse<TResponse>> CreateGraphQLRequestObservable<TR
             .Where(response => response != null)
             .TakeUntil(response => response.Type == GraphQLWebSocketMessageType.GQL_CONNECTION_ACK ||
                                    response.Type == GraphQLWebSocketMessageType.GQL_CONNECTION_ERROR)
-            .LastAsync()
+            .LastOrDefault()
```

Comment:
Did system.reactive's LastAsync have OrDefault behaviour baked in?

I see on line 251 we dereference `response.Type` (the result of ackTask).  If that's null, can we get an NRE here?

Reply:
I would guess .Last is probably closer?

Reply:
Might be some blocking/perf implications with these kinds of workarounds?   I chatted with chatgpt:

https://scopeardev.slack.com/archives/GCK7VBW2F/p1695074890022519?thread_ts=1694794181.707849&cid=GCK7VBW2F

Some tl;dr from that:

> So, while both methods will wait for the last value, using .Last() on an asynchronous observable should be done cautiously, considering the potential performance and thread-blocking implications, especially in scenarios where responsiveness and concurrency are important. If you want to ensure non-blocking behavior with asynchronous observables, it's recommended to use .LastAsync().